lib/recurly/addon.php

```diff
@@ -55,8 +55,8 @@ protected function uri() {
       return Recurly_Addon::uriForAddOn($this->plan_code, $this->add_on_code);
   }
   protected static function uriForAddOn($planCode, $addonCode) {
-    return (Recurly_Client::PATH_PLANS . '/' . rawurlencode($planCode) .
-            Recurly_Client::PATH_ADDONS . '/' . rawurlencode($addonCode));
+    $path = self::_uriForResource(Recurly_Client::PATH_ADDONS, rawurlencode($addonCode));
+    return (Recurly_Client::PATH_PLANS . '/' . rawurlencode($planCode) .$path);
```

Comment:
Could we perhaps make `uriForResource` take varargs? then it can handle as many parameters as you have. Seems weird to break out of the pattern here.

Reply:
Great idea.